src/ir/type-updating.h

```diff
@@ -434,13 +439,23 @@ class TypeMapper : public GlobalTypeRewriter {
     : GlobalTypeRewriter(wasm), mapping(mapping) {}
 
   void map() {
-    // Map the types of expressions (curr->type, etc.) to their merged
-    // types.
-    mapTypes(mapping);
-
     // Update the internals of types (struct fields, signatures, etc.) to
     // refer to the merged types.
-    update();
+    auto oldToNewTypes = rebuildTypes();
+
+    // Compose the given mapping from old types to old types with the new
+    // mapping from old types to new types.
```

Comment:
Maybe add some clarification as to how the composition works between `oldToNewTypes` and `mapping`? Something like `oldToNewTypes[mapping[x]] = y` or such.